### ISSUE TITLE
[Crash]: GS trying to get available railtypes of an invalid company

### ISSUE THREAD

**Issue by SamuXarick** [OpenTTD/OpenTTD#10407]
### Version of OpenTTD

13.0-RC1, Windows 11 Pro

### Steps to reproduce

Code a GS with this inside function Start()
```
		local railtypes;
		GSCompanyMode(0) && (railtypes = GSRailTypeList());
```

Start a new game with this GS.

### Upload crash files

[crash20230124223021.zip](https://github.com/OpenTTD/OpenTTD/files/10494545/crash20230124223021.zip)

**Comment by glx22** [comment]
I'm not sure this syntax is correct for using GSCompanyMode().

**Comment by James103** [comment]
The actual problem as far as I can tell is that `GSCompanyMode` can be called with invalid companies, which can lead to other problems and crashes, not just this specific one.

**Comment by SamuXarick** [comment]
This was fixed with #10529

![image](https://github.com/OpenTTD/OpenTTD/assets/43006711/ebb891e1-c452-4dfe-9e3d-4bbc6679c987)
